File: api/add_contact.php
<?php
include 'conexion.php';

// Obtiene los datos del nuevo contacto desde la solicitud POST.
$name = $_POST['name'];
$surname = $_POST['surname'];
$phone = $_POST['phone'];

// Inserta el nuevo contacto en la base de datos.
$sql = "INSERT INTO contacts (name, surname, phone) VALUES ('$name', '$surname', '$phone')";

if ($connect->query($sql) === true) {
	echo "Contacto añadido correctamente";
} else {
	echo "Error al añadir contacto: " . $connect->error;
}

// Cierra la conexión.
$connect->close();

File: lib/add_contact.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

Future<void> addContact(String name, String surname, String phone) async {
  try {
    final response = await http.post(
      Uri.parse('http://10.0.0.32:8080/contacts/add_contact.php'),
      body: {
        'name': name,
        'surname': surname,
        'phone': phone,
      },
    );

    if (response.statusCode == 200) {
      print("Contacto añadido correctamente");
      
    } else {
      throw Exception(
          'Failed to add contact. Status code: ${response.statusCode}');
    }
  } catch (error) {
    print('Error al añadir el contacto: $error');
  }
}
